docs/source/contributor-guide/index.md

```diff
@@ -66,24 +66,33 @@ ideas with the community to get feedback on implementation.
 
 ## Pull Request Overview
 
-We welcome pull requests (PRs) from anyone from the community.
+We welcome pull requests (PRs) from anyone in the community.
 
-DataFusion is a very active fast-moving project and we try to review and merge PRs quickly to keep the review backlog down and the pace up. After review and approval, one of the [committers] will merge your PR.
+DataFusion is a very active fast-moving project and we try to review and merge PRs quickly.
 
 Review bandwidth is currently our most limited resource, and we highly encourage reviews by the broader community. If you are waiting for your PR to be reviewed, consider helping review other PRs that are waiting. Such review both helps the reviewer to learn the codebase and become more expert, as well as helps identify issues in the PR (such as lack of test coverage), that can be addressed and make future reviews faster and more efficient.
 
+The lifecycle of a PR is:
+
+1. Create a PR targeting the `main` branch.
+2. For returning contributors, CI tests are automatically run. For new contributors a committer must first trigger the tests.
+3. Your PR will be reviewed. Please respond to all feedback on the PR: you don't have to change the code, but you should acknowledge the feedback.
```

Comment:
```suggestion
3. Please respond to all feedback on the PR review: you don't have to change the code, but you should acknowledge the feedback. PRs waiting for the feedback for more than days will be moved to draft.
```

Reply:
in https://github.com/apache/datafusion/pull/10532/commits/e70f0291ac4b14592a6bec51dcd365f30d1246f2